modules/tpmnBidAdapter.js

```diff
@@ -131,34 +227,91 @@ function createSite(refInfo) {
   return site;
 }
 
-function parseSize(size) {
-  let sizeObj = {}
-  sizeObj.width = parseInt(size[0], 10);
-  sizeObj.height = parseInt(size[1], 10);
-  return sizeObj;
-}
+function buildVideoImp(bidRequest, imp) {
+  const videoAdUnitParams = utils.deepAccess(bidRequest, `mediaTypes.${VIDEO}`, {});
+  const videoBidderParams = utils.deepAccess(bidRequest, `params.${VIDEO}`, {});
 
-function parseSizes(sizes) {
-  if (Array.isArray(sizes[0])) { // is there several sizes ? (ie. [[728,90],[200,300]])
-    return sizes.map(size => parseSize(size));
+  const videoParams = { ...videoAdUnitParams, ...videoBidderParams };
+
+  const videoSizes = (videoAdUnitParams && videoAdUnitParams.playerSize) || [];
+
+  if (videoSizes && videoSizes.length > 0) {
+    utils.deepSetValue(imp, 'video.w', videoSizes[0][0]);
+    utils.deepSetValue(imp, 'video.h', videoSizes[0][1]);
   }
-  return [parseSize(sizes)]; // or a single one ? (ie. [728,90])
-}
 
-function getBannerSizes(bidRequest) {
-  return parseSizes(deepAccess(bidRequest, 'mediaTypes.banner.sizes') || bidRequest.sizes);
+  VIDEO_ORTB_PARAMS.forEach((param) => {
```

Comment:
this seems unnecessary; the ortb convertor is already doing all this for you. Is that correct @dgirardi ?

Reply:
i think.. this is needed to check which parameters are allowed.

Reply:
I think [this example](https://github.com/prebid/Prebid.js/blob/master/libraries/ortbConverter/README.md#example-taking-video-parameters-from-bidrequestparamsvideo) does this with less code, as long as the parameters you're checking match the [ones used in ortbConverter](https://github.com/prebid/Prebid.js/blob/4a3fcf5b1fe843f17ba6ba19d9c5549dacf44f2f/libraries/ortbConverter/processors/video.js#L6-L29)

Reply:
[2699115](https://github.com/prebid/Prebid.js/pull/10611/commits/2699115a76ed891ec7f911ad924206b399790ad9)

I modified the code in the direction you suggested. ([this example](https://github.com/prebid/Prebid.js/blob/master/libraries/ortbConverter/README.md#example-taking-video-parameters-from-bidrequestparamsvideo) )